pkg/controller/nodes/executor.go

```diff
@@ -374,6 +374,7 @@ func (c *nodeExecutor) handleNode(ctx context.Context, w v1alpha1.ExecutableWork
 			return executors.NodeStatusUndefined, err
 		}
 
+		nodeStatus.IncrementAttempts()
```

Comment:
Can you add docs for why we should do it here and not after the handle() returns?

Reply:
yup